### PR TITLE
Reconfigure actions & upload image to Docker Hub

### PR DESCRIPTION
**Pull request by domanskyi**
`Build image, run tests on Docker and push image to the Docker registry after PR approved and code merged to master branch. Image tag should contain a commit tag. + Update README.md`

## Changes:
- added  README
- updated listing and tests action to run on push
- added action to publish Docker image generated from PR. [Generated images](https://hub.docker.com/repository/docker/domanskyi/random-frog-api/tags?page=1&ordering=last_updated)
- remove redundant packages

### CODE REVIEW

File: .github/workflows/publish-image.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - uses: actions/checkout@v2
+
+      # https://github.com/docker/metadata-action - generate Docker meta
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ secrets.DOCKERHUB_USERNAME }}/random-frog-api
+          flavor: |
+            latest=true
```
Comment (domanskyi) [inline .github/workflows/publish-image.yml:24]: Not sure that I should also always regenerate `latest` tag

File: Dockerfile [hunk 2]
```diff
@@ -1,9 +1,13 @@
-FROM node:14-alpine
+FROM node:14-alpine 
 
 WORKDIR /usr/src/random-frog-api
 
 COPY . .
 
+RUN npm ci
+RUN npm run test
```
Comment (domanskyi) [inline Dockerfile:8]: `Build image, run tests on Docker` 
Should tests be runned as part of the action or here?

File: .github/workflows/publish-image.yml (hunk 1 above)
Comment (m1g0r) [reply .github/workflows/publish-image.yml:24]: yes, this is a good approach provided it is the master/main branch
here you can add some check and add this tag only master/main
for other branches, you can add another tag like a branch name with a short SHA

File: Dockerfile (hunk 2 above)
Comment (m1g0r) [reply Dockerfile:8]: yes, this solution can be used

**Pull request merged**